### PR TITLE
Fix frozen layers

### PR DESCRIPTION
**Pull request by Enealor**
Test to replicate the issue: [link](https://github.com/Enealor/PyTorch-SM3/blob/066df21f14c181cf6bdf569a6883e2986ebbf60d/tests/test_SM3.py#L6-L31)

Fix: Check if the current Parameter has a grad attribute equal to None (the default value).

### CODE REVIEW

**Comment by Enealor** [comment]
Changes: Had to update the test because the "simple loss function" I used is not available in Torch 1.4.
https://github.com/Enealor/PyTorch-SM3/blob/d872ed4b9fa6e15f4d36a0e0e2eeb62c40b00867/tests/test_SM3.py#L7-L29 is the new test.
